lib/slack.ml

```diff
@@ -418,7 +434,24 @@ let generate_status_notification ?slack_user_id ?failed_steps ~(ctx : Context.t)
         failed_steps
   in
   let attachment = { empty_attachments with mrkdwn_in = Some [ "fields"; "text" ]; color = Some color_info; text } in
-  make_message ~text:summary ~attachments:[ attachment ] ~channel:(Status_notification.to_slack_channel channel) ()
+  let reply_of_log log = log |> Text_cleanup.cleanup |> String.split_on_char '\r' |> String.concat "\n" in
+  let handler (upload_file : file:file_req -> (unit, string) result Lwt.t) (res : Slack_t.post_message_res) :
+    (unit, _) result Lwt.t =
+    match job_log with
+    | [] -> Lwt.return_ok ()
+    | _ :: _ ->
+      let ({ ts; channel; _ } : post_message_res) = res in
+      let files =
+        job_log
+        |> List.map (fun (job_name, job_log) ->
+               { name = job_name; title = None; alt_txt = None; content = reply_of_log job_log })
+      in
+      let file = { files; channel_id = Some channel; initial_comment = None; thread_ts = Some ts } in
+      upload_file ~file
```

Comment:
most of this logic should be handled in the `action` module. No need to pass the failed steps logs to the `generate_status_notification` fn, you can just call the upload file with the channel and thread ts values. 

also `upload_file` probably isn't the correct name for the function